src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultAssemblyPartDiscoveryProvider.cs

```diff
@@ -101,23 +100,28 @@ private Dependency CreateDependency(RuntimeLibrary library, ISet<string> referen
 
             private DependencyClassification ComputeClassification(string dependency)
             {
-                Debug.Assert(_dependencies.ContainsKey(dependency));
+                if (!_runtimeDependencies.ContainsKey(dependency))
```

Comment:
Also filed https://github.com/aspnet/MusicStore/issues/757, because I think the discovery might be broken if Mvc is referenced thru a metapackage which is something that's planned.